### PR TITLE
Throw error on connection close

### PR DESCRIPTION
**Pull request by musgravejw**
Throw an error in subscribe method when AMQP connection is closed.

### CODE REVIEW

File: lib/subscribe.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ module.exports = function Subscribe(url, opts) {
     Channel(url, opts, function (err, ch, conn) {
       if (err) throw err;
 
+      conn.on('close', function () {
```
Comment (theworkflow) [inline lib/subscribe.js]: http://www.squaremobius.net/amqp.node/channel_api.html#channel_close

Comment (theworkflow) [inline lib/subscribe.js]: That function throws an error. You can return that.

Comment (jackboberg) [inline lib/subscribe.js]: are your recommending `on('error', fn)` instead?

Comment (theworkflow) [inline lib/subscribe.js]: I used wrong wording. I believe throwing the error is the right thing (for now)

File: lib/subscribe.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ module.exports = function Subscribe(url, opts) {
     Channel(url, opts, function (err, ch, conn) {
       if (err) throw err;
 
+      conn.on('close', function (err) {
```
Comment (theworkflow) [inline lib/subscribe.js]: Supposed to be `ch.on('close', fn)`

**Pull request closed without merging**